src/main/java/org/opensearch/ad/rest/RestStatsAnomalyDetectorAction.java

```diff
@@ -11,64 +11,150 @@
         return STATS_ANOMALY_DETECTOR_ACTION;
     }
 
     @Override
-    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
+    public List<RouteHandler> routeHandlers() {
+        return ImmutableList
+            .of(
+                new RouteHandler(
+                    RestRequest.Method.GET,
+                    String.format(Locale.ROOT, "%s/%s/{%s}/%s", AnomalyDetectorExtension.AD_BASE_DETECTORS_URI, "v1", NODE_ID, "stats"),
+                    handleRequest
+                ),
+                new RouteHandler(
+                    RestRequest.Method.GET,
+                    String
+                        .format(
+                            Locale.ROOT,
+                            "%s/%s/{%s}/%s/{%s}",
+                            AnomalyDetectorExtension.AD_BASE_DETECTORS_URI,
+                            "v1",
+                            NODE_ID,
+                            "stats",
+                            STAT
+                        ),
+                    handleRequest
+                ),
+                new RouteHandler(RestRequest.Method.GET, AnomalyDetectorExtension.AD_BASE_DETECTORS_URI + "/stats", handleRequest),
+                new RouteHandler(
+                    RestRequest.Method.GET,
+                    String.format(Locale.ROOT, "%s/%s/{%s}", AnomalyDetectorExtension.AD_BASE_DETECTORS_URI, "stats", STAT),
+                    handleRequest
+                )
+            );
+    }
+
+    private Function<RestRequest, ExtensionRestResponse> handleRequest = (request) -> {
+        try {
+            return prepareRequest(request);
+        } catch (Exception e) {
+            // TODO: handle the AD-specific exceptions separately
+            return exceptionalRequest(request, e);
+        }
+    };
+
+    protected ExtensionRestResponse prepareRequest(RestRequest request) throws IOException {
         if (!EnabledSetting.isADPluginEnabled()) {
             throw new IllegalStateException(CommonErrorMessages.DISABLED_ERR_MSG);
         }
         ADStatsRequest adStatsRequest = getRequest(request);
-        return channel -> client.execute(StatsAnomalyDetectorAction.INSTANCE, adStatsRequest, new RestToXContentListener<>(channel));
+        CompletableFuture<StatsAnomalyDetectorResponse> statsFutureResponse = new CompletableFuture<>();
+        sdkRestClient
+            .execute(
+                StatsAnomalyDetectorAction.INSTANCE,
+                adStatsRequest,
+                ActionListener.wrap(response -> statsFutureResponse.complete(response), ex -> statsFutureResponse.completeExceptionally(ex))
+            );
+
+        StatsAnomalyDetectorResponse statsAnomalyDetectorResponse = statsFutureResponse
+            .orTimeout(AnomalyDetectorSettings.REQUEST_TIMEOUT.get(settings).getMillis(), TimeUnit.MILLISECONDS)
+            .join();
+
+        logger.info("Stats Response is  **********************  " + statsAnomalyDetectorResponse.toString());
```

Comment:
We don't have to log the response here. Leftover?

Reply:
yes. Sorry it was left by mistake. Will remove it in the new iteration.